File: AI_Persona/backend/generate_token.py
#!/usr/bin/env python3
"""
Simple LiveKit Token Generator
Generates participant tokens for testing
"""

import jwt
import time
import os
from dotenv import load_dotenv

load_dotenv()

def generate_token(room_name="interview-room", participant_name="candidate-1"):
    """Generate a LiveKit participant token using PyJWT"""
    
    api_key = os.getenv("LIVEKIT_API_KEY")
    api_secret = os.getenv("LIVEKIT_API_SECRET")
    
    if not api_key or not api_secret:
        print("❌ Error: API keys not found in .env")
        return None
    
    # Token payload
    payload = {
        "exp": int(time.time()) + 86400,  # 24 hours
        "iss": api_key,
        "nbf": int(time.time()),
        "sub": participant_name,
        "name": participant_name,
        "video": {
            "room": room_name,
            "roomJoin": True,
            "canPublish": True,
            "canSubscribe": True,
        }
    }
    
    # Generate token
    token = jwt.encode(payload, api_secret, algorithm="HS256")
    
    print("✅ Token Generated Successfully!")
    print("")
    print(f"Room: {room_name}")
    print(f"Participant: {participant_name}")
    print(f"Valid for: 24 hours")
    print("")
    print("=" * 60)
    print("TOKEN:")
    print("=" * 60)
    print(token)
    print("=" * 60)
    print("")
    print("📋 Copy the token above and paste it into test.html")
    
    return token

if __name__ == "__main__":
    import sys
    room = sys.argv[1] if len(sys.argv) > 1 else "interview-room"
    participant = sys.argv[2] if len(sys.argv) > 2 else "candidate-1"
    generate_token(room, participant)

File: AI_Persona/backend/src/generate-token.ts
import { AccessToken } from 'livekit-server-sdk';
import dotenv from 'dotenv';
import { getDefaultUserProfile } from './interviewers/index.js';
import type { RoomMetadata } from './types/interviewer.js';

// Load environment variables
dotenv.config();

const apiKey = process.env.LIVEKIT_API_KEY;
const apiSecret = process.env.LIVEKIT_API_SECRET;

if (!apiKey || !apiSecret) {
    console.error('❌ Error: LIVEKIT_API_KEY and LIVEKIT_API_SECRET must be set in .env file');
    process.exit(1);
}

// Parse CLI arguments
const args = process.argv.slice(2);
const interviewerId = args[0] || 'frontend';
const roomName = 'interview-room';
const participantName = 'candidate-1';

// Get default user profile
const userProfile = getDefaultUserProfile();

// Create room metadata
const metadata: RoomMetadata = {
    interviewerId,
    userProfile,
};

// Create access token
const token = new AccessToken(apiKey, apiSecret, {
    identity: participantName,
    name: participantName,
    metadata: JSON.stringify(metadata), // Add metadata to token
});

// Add room permissions
token.addGrant({
    room: roomName,
    roomJoin: true,
    canPublish: true,
    canSubscribe: true,
});

// Generate JWT
const jwt = await token.toJwt();

// Output
console.log('✅ Token Generated Successfully!\n');
console.log(`Room: ${roomName}`);
console.log(`Participant: ${participantName}`);
console.log(`Interviewer: ${interviewerId}`);
console.log(`Candidate: ${userProfile.name} (${userProfile.experience})`);
console.log(`Valid for: 24 hours\n`);
console.log('============================================================');
console.log('TOKEN:');
console.log('============================================================');
console.log(jwt);
console.log('============================================================\n');
console.log('📋 Copy the token above and paste it into test.html');
console.log('\n💡 Tip: Generate token for different interviewer:');
console.log('   pnpm run token backend');
console.log('   pnpm run token system-design');
